### PR TITLE
Video Front-end

### PR DESCRIPTION
**Pull request by rMazeiks**
Hi @alfredsgenkins,

Can you please take a look at this implementation of video support? Thanks!

This branch adds support for YouTube and Vimeo videos in the product detail page (SWPWA-250).
- Updated queries to fetch video thumbnail data
- Installed react libraries for handling YouTube and Vimeo videos
- Refactored media gallery to allow for different types of media
- Created VideoThumbnail component, output it in the media gallery carousel
- Created VideoPopup component for video playback

### CODE REVIEW

File: src/app/component/VideoPopup/VideoPopup.component.js [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+ * See LICENSE for license details.
+ *
+ * @license OSL-3.0 (Open Software License ("OSL") v. 3.0)
+ * @package scandipwa/base-theme
+ * @link https://github.com/scandipwa/base-theme
+ */
+
+import { PureComponent } from 'react';
+import Vimeo from 'react-vimeo';
+import YouTube from 'react-youtube';
```
Comment (alfredsgenkins) [inline src/app/component/VideoPopup/VideoPopup.component.js]: Please load them async, I do not want 20kb+ additional code in my bundle

File: src/app/component/VideoThumbnail/VideoThumbnail.component.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+    /**
+     * Renders an icon indicating that the video can be played
+     */
+    renderPlayIcon() {
+        return (
+            <div block="VideoThumbnail" elem="IconContainer">
+                <span block="VideoThumbnail" elem="PlayIcon">
+                    { /* eslint-disable-next-line */ }
+                    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M8 5v14l11-7z"/><path d="M0 0h24v24H0z" fill="none"/></svg>
```
Comment (alfredsgenkins) [inline src/app/component/VideoThumbnail/VideoThumbnail.component.js]: Please draw this with CSS

File: src/app/query/ProductList.query.js [hunk 3]
```diff
@@ -312,9 +322,37 @@ export class ProductListQuery {
...
+     */
+    _getMediaThumbnailField() {
+        return new Field('thumbnail').addField('url');
+    }
+
     _getMediaGalleryField() {
         return new Field('media_gallery_entries')
-            .addFieldList(this._getMediaGalleryFields());
+            .addFieldList(this._getMediaGalleryFields())
+            .addField(this._getVideoContentField())
```
Comment (alfredsgenkins) [inline src/app/query/ProductList.query.js]: Please use only `addFieldList` and separate every field creating into a function

Comment (alfredsgenkins) [reply src/app/query/ProductList.query.js]: Reference productList query for example

**Comment by alfredsgenkins** [comment]
Fix conflicts & CR issues, otherwise I will

**Comment by Danpiel** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/app/component/VideoThumbnail/VideoThumbnail.container.js  1
- src/app/component/VideoPopup/VideoPopup.container.js  2
- src/app/component/VideoPopup/VideoPopup.component.js  5
- src/app/component/VideoThumbnail/VideoThumbnail.component.js  1
- src/app/component/ProductGallery/ProductGallery.component.js  2
         

Complexity decreasing per file
==============================
+ src/app/component/ProductGallery/ProductGallery.container.js  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/ScandiPWA/base-theme/pullRequest?prid=4444462&bid=15029175)

**Comment by Danpiel** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/app/component/VideoPopup/VideoPopup.container.js  2
- src/app/component/VideoThumbnail/VideoThumbnail.container.js  1
- src/app/component/VideoPopup/VideoPopup.component.js  5
- src/app/component/ProductGallery/ProductGallery.component.js  2
- src/app/component/VideoThumbnail/VideoThumbnail.component.js  1
         

Complexity decreasing per file
==============================
+ src/app/component/ProductGallery/ProductGallery.container.js  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/ScandiPWA/base-theme/pullRequest?prid=4444462&bid=15029175)

**Comment by rMazeiks** [comment]
Hi @alfredsgenkins,

Thanks for the feedback - I have updated the PR as you have asked. I think I managed async imports to load correctly, but I noticed errors being output in the console (even though the page is working):

attels

I'm not sure what could be causing this but I think it has something to do with the async loading and the service worker, but I couldn't figure out what the problem was... Could you please check if I made the async imports correctly?

**Comment by alfredsgenkins** [comment]
Merged in https://github.com/scandipwa/base-theme/pull/375

**Pull request closed without merging**